src/nnet3/nnet-descriptor.h

```diff
@@ -192,6 +198,12 @@ class SwitchingForwardingDescriptor: public ForwardingDescriptor {
   SwitchingForwardingDescriptor(std::vector<ForwardingDescriptor*> &src):
       src_(src) { }
   virtual ~SwitchingForwardingDescriptor() { DeletePointers(&src_); }
+
+  // defined in nnet-utils.h
+  friend void ExtractDescriptorValuesFromForwardingDescriptor(const Nnet &Nnet,
```

Comment:
Also DescriptorValues is not sufficiently clear. 

@danpovey Yiming uses this function to extract things like the offsets in the Offset descriptor or the t-modulus value in the RoundDescriptor. Could you suggest a name which might be descriptive enough for these functions ?